lib/filesystem.nix

```
@@ -2,12 +2,94 @@
 { lib }:
 
 let
+  inherit (builtins)
+    readDir
+    pathExists
+    ;
+
   inherit (lib.strings)
     hasPrefix
     ;
+
+  inherit (lib.filesystem)
+    pathType
+    ;
 in
 
 {
+
+  /*
+    The type of a path. The path needs to exist and be accessible.
+    The result is either "directory" for a directory, "regular" for a regular file, "symlink" for a symlink, or "unknown" for anything else.
+
+    Type:
+      pathType :: Path -> String
+
+    Example:
+      pathType /.
+      => "directory"
+
+      pathType /some/file.nix
+      => "regular"
+  */
+  pathType =
+    # Introduced in Nix 2.14
+    builtins.readFileType or
+    (path:
```

```suggestion
    builtins.readFileType or
    # Nix <2.14 compatibility shim
    (path:
```

Same info, but more useful to a sporadic user.